ENG-11059: Adds section about Kueue alerts (#451)
opendatahub-io/opendatahub-documentation: modules/viewing-kueue-alerts-for-distributed-workloads.adoc

endif::[]

ifndef::upstream[]
* You can access a data science project that contains a workbench, and the workbench is running a default notebook image that contains the CodeFlare SDK, for example, the *Standard Data Science* notebook. 
For information about projects and workbenches, see link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects[Working on data science projects].
endif::[]
ifdef::upstream[]
* You can access a data science project that contains a workbench, and the workbench is running a default notebook image that contains the CodeFlare SDK, for example, the *Standard Data Science* notebook. 
For information about projects and workbenches, see link:{odhdocshome}/working-on-data-science-projects[Working on data science projects].
endif::[]

* You have logged in to {productname-long}.
* Your data science project contains distributed workloads.

.Procedure

ifdef::upstream,self-managed[]
. In the {openshift-platform} console, click *Observe* -> *Alerting*.
endif::[]
ifdef::cloud-service[]
. In the OpenShift console, click *Observe* -> *Alerting*.
endif::[]

> 

Reviewer: Complete path should be mentioned here to reach "Alerting rules tab" like
Click Observe → Alerting and then click the Alerting rules tab

Author: The first part of the path is provided in the previous step -- see lines 41-46:
> In the OpenShift console, click *Observe* -> *Alerting*.

Reviewer: I viewed this file and it doesn't show 

![Screenshot from 2024-09-12 16-48-34](https://github.com/user-attachments/assets/e257047c-96d4-4400-b057-f162f56e36e5)

Reviewer: For some reason, it doesn't appear in preview since the statement is present, so am fine with this

Author: Conditionalised text doesn't show in generated previews in Git.

Author: Thank you @ChughShilpa !

> . Click the *Alerting rules* tab to view a list of all alerting rules. 

Reviewer 2: ```suggestion
. Click the *Alerting rules* tab. 
```
Feels repetitive - can we use a different word to describe alerting rules or remove the repetition? "Click the alerting rules tab to view alerts that match your defined rules" ?

Author: Added descriptive text.

* The *Severity* column indicates whether the alert is informational, a warning, or critical.

> * The *Alert state* column indicates whether a rule is currently firing.

Reviewer 2: ```suggestion
* The *Alert state* column indicates whether a rule is currently active.
```
A bit nitpicky so I don't mind if you toss this one - "firing" has a slightly negative connotation and feels a bit colloquial.

Reviewer 2: I do see that this is used in the UI - I'm happy to set this aside if you need to reflect the UI. If you decide to go with this change, there are a few other instances later in the doc that should be updated as well.

Author: Thank you @eturner24. I had a similar concern but felt it was important to be consistent with the UI and with the OpenShift documentation, where the term "active" is used for the "Pending" state, and the term "firing" is used for the "Firing" state.

> . Click the name of an alerting rule to see more details about what causes the rule to fire.  

Reviewer 2: ```suggestion
. Click the name of an alerting rule to see more details about what activates the rule.  
```
Same comment as previous

Author: Rephrased.

The alerting rules for Kueue resources are listed in the following table.
+
.Alerting rules for Kueue resources
[cols="15,20,65"]
|===

> |Severity | Name | Fires when ...

Reviewer 2: ```suggestion
|Severity | Name | Alert Condition
```
Also suggest rewriting this if there's a potential for localization in the future - it's hard to translate cells that finish a header sentence into other languages

Author: Updated as suggested, but used sentence-style capitalisation for the column heading as advised in _IBM Style_.

|*Critical*
|`KueuePodDown`
|The Kueue pod is not ready for a period of 5 minutes.

|*Info*
|`LowClusterQueueResourceUsage`
|Resource usage in the cluster queue is below 20% of its nominal quota for more than 1 day. 
Resource usage refers to any resources listed in the cluster queue, such as CPU, memory, and so on.

|*Info*
|`ResourceReservationExceedsQuota`
|Resource reservation is 10 times the available quota in the cluster queue. 
Resource reservation refers to any resources listed in the cluster queue, such as CPU, memory, and so on.

|*Info*
|`PendingWorkloadPods`
|A pod has been in a `Pending` state for more than 3 days.

|===

. If the *Alert state* of an alerting rule is set to *Firing*, complete the following steps:

> .. Click *Observe* -> *Alerting* and then click the *Alerts* tab. 

Reviewer: If you we are mentioning this path above as suggested, then it can be removed from here.
Simply can mention 
Click "Alerts tab"

Author: If the user has followed all of the steps and clicked a rule name to see more details, they can no longer see the Alerts tab and must click the full path as specified.
